packages/pcre/SPECS/pcre.spec

```diff
@@ -0,0 +1,930 @@
+# This amends ABI, application built with this patch cannot run with
+# previous libpcreposix builds.
+Patch3:     pcre-8.42-Declare-POSIX-regex-function-names-as-macros-to-PCRE.patch
+# Add (*LF) to a list of start-of-pattern options in the C++ wrapper,
+# upstream bug #2400, in upstream after 8.43
+Patch4:     pcre-8.43-Fix-omission-of-LF-from-list-in-the-C-wrapper.patch
+
+Patch100:	pcre0843.irix.patch
+
+BuildRequires:  readline-devel
+BuildRequires:  autoconf
+BuildRequires:  automake
+BuildRequires:  coreutils
+BuildRequires:  gcc
+BuildRequires:  gcc-c++
+#BuildRequires:  glibc-common
+BuildRequires:  libtool
+BuildRequires:  make
+# perl not used because config.h.generic is pregenerated
+# Tests:
+BuildRequires:  bash
+BuildRequires:  diffutils
+BuildRequires:  grep
+
+%description
+PCRE, Perl-compatible regular expression, library has its own native API, but
+a set of wrapper functions that are based on the POSIX API are also supplied
+in the libpcreposix library. Note that this just provides a POSIX calling
+interface to PCRE: the regular expressions themselves still follow Perl syntax
+and semantics. This package provides support for strings in 8-bit and UTF-8
+encodings. Detailed change log is provided by %{name}-doc package.
+
+%package utf16
+Summary:    UTF-16 variant of PCRE
+Conflicts:  %{name}%{?_isa} < 8.38-12
+
+%description utf16
+This is Perl-compatible regular expression library working on UTF-16 strings.
+Detailed change log is provided by %{name}-doc package.
+
+%package utf32
+Summary:    UTF-32 variant of PCRE
+Conflicts:  %{name}%{?_isa} < 8.38-12
+
+%description utf32
+This is Perl-compatible regular expression library working on UTF-32 strings.
+Detailed change log is provided by %{name}-doc package.
+
+%package cpp
+Summary:    C++ bindings for PCRE
+Requires:   %{name}%{?_isa} = %{version}-%{release}
+
+%description cpp
+This is C++ bindings for the Perl-compatible regular expression library.
+Detailed change log is provided by %{name}-doc package.
+
+%package doc
+Summary:    Change log for %{name}
+BuildArch:  noarch
+
+%description doc
+These are large documentation files about PCRE.
+
+%package devel
+Summary:    Development files for %{name}
+Requires:   %{name}%{?_isa} = %{version}-%{release}
+Requires:   %{name}-cpp%{?_isa} = %{version}-%{release}
+Requires:   %{name}-utf16%{?_isa} = %{version}-%{release}
+Requires:   %{name}-utf32%{?_isa} = %{version}-%{release}
+
+%description devel
+Development files (Headers, libraries for dynamic linking, etc) for %{name}.
+
+%package static
+Summary:    Static library for %{name}
+Requires:   %{name}-devel%{_isa} = %{version}-%{release}
+
+%description static
+Library for static linking for %{name}.
+
+%package tools
+Summary:    Auxiliary utilities for %{name}
+Requires:   %{name}%{_isa} = %{version}-%{release}
+
+%description tools
+Utilities demonstrating PCRE capabilities like pcregrep or pcretest.
+
+%prep
+%setup -q -n %{name}-%{myversion}
+# Get rid of rpath
```

Comment:
For the rse - we don't want to get rid of rpath - and we'll need to check if the resulting library still has it.

Normally you can see this with `objdump -p /PATH/TO/libblah.so |grep RPATH`

Reply:
This is how it is set in the original spec. A better view of what's changed is:

`diff -Naur SPECS/pcre.spec.origfedora SPECS/pcre.spec
--- SPECS/pcre.spec.origfedora  2020-02-10 08:22:03.734613629 +0000
+++ SPECS/pcre.spec     2020-02-10 13:47:23.418975365 +0000
@@ -43,13 +43,16 @@
 # Add (*LF) to a list of start-of-pattern options in the C++ wrapper,
 # upstream bug #2400, in upstream after 8.43
 Patch4:     pcre-8.43-Fix-omission-of-LF-from-list-in-the-C-wrapper.patch
+
+Patch100:      pcre0843.irix.patch
+
 BuildRequires:  readline-devel
 BuildRequires:  autoconf
 BuildRequires:  automake
 BuildRequires:  coreutils
 BuildRequires:  gcc
 BuildRequires:  gcc-c++
-BuildRequires:  glibc-common
+#BuildRequires:  glibc-common
 BuildRequires:  libtool
 BuildRequires:  make
 # perl not used because config.h.generic is pregenerated
@@ -129,15 +132,17 @@
 %patch2 -p2
 %patch3 -p1
 %patch4 -p1
+
+%patch100 -p1
 # Because of rpath patch
 libtoolize --copy --force
 autoreconf -vif
 # One contributor's name is non-UTF-8
-for F in ChangeLog; do
-    iconv -f latin1 -t utf8 "$F" >"${F}.utf8"
-    touch --reference "$F" "${F}.utf8"
-    mv "${F}.utf8" "$F"
-done
+#for F in ChangeLog; do
+#    iconv -f latin1 -t utf8 "$F" >"${F}.utf8"
+#    touch --reference "$F" "${F}.utf8"
+#    mv "${F}.utf8" "$F"
+#done

 %build
 # There is a strict-aliasing problem on PPC64, bug #881232
@@ -173,10 +178,10 @@
 %endif
 make %{?_smp_mflags} check VERBOSE=yes

-%ldconfig_scriptlets
-%ldconfig_scriptlets utf16
-%ldconfig_scriptlets utf32
-%ldconfig_scriptlets cpp
+#%ldconfig_scriptlets
+#%ldconfig_scriptlets utf16
+#%ldconfig_scriptlets utf32
+#%ldconfig_scriptlets cpp

 %files
 %{_libdir}/libpcre.so.1`

Reply:
I've updated the spec with one that works with latest grep.